### PR TITLE
refactor(shared): make shared click-outside hook

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Move the click-outside code to a shared hook.
- Update shared workspace to using consistent linting.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the config tab in machine details.
- Click on the tag form Edit button.
- Click on the tag input and the dropdown should appear.
- Click outside the dropdown and it should close.

## Fixes

Fixes: canonical-web-and-design/app-tribe#835.

### CODE REVIEW

File: shared/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+module.exports = {
```
Comment (huwshimi) [inline shared/.eslintrc.js:1]: This is pretty much just copied and pasted from the ui workspace.

File: shared/src/components/Footer/Footer.test.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
+import React from "react";
```
Comment (huwshimi) [inline shared/src/components/Footer/Footer.test.tsx:1]: Most of these files are just reordering the imports etc. to satisfy eslint.

File: ui/src/app/base/components/TagSelector/TagSelector.tsx [hunk 3]
```diff
@@ -257,63 +242,63 @@ export const TagSelector = ({
...
-            if (e.key === "Enter") {
-              e.preventDefault();
-              if (allowNewTags) {
-                const cleanedFilter = sanitiseFilter(filter);
-                if (onAddNewTag) {
-                  onAddNewTag(cleanedFilter.name);
-                  setFilter("");
-                } else {
-                  updateTags([...selectedTags, cleanedFilter]);
+    <div id={wrapperId} ref={wrapperRef}>
```
Comment (huwshimi) [inline ui/src/app/base/components/TagSelector/TagSelector.tsx:245]: Below this are all indentation changes caused by adding this wrapping div.

**Review by huwshimi — COMMENTED** [review]
I've pulled out all these changes from my branch where I've implemented the new Vanilla navigation. Sorry if it's a bit of a mess, hopefully it mostly makes sense.

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Thanks for the comments - made reviewing the code much easier

**Pull request merged**